sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -240,6 +240,32 @@ public void testInWithSubqueryOrderImplicitlyAppliedWithExistingOrderByOnDiffere
                                    "Collect[doc.t1 | [x, a] | All]\n"));
     }
 
+    @Test
+    public void testParentQueryIsPushedDownAndMergedIntoSubRelationWhereClause() {
+        LogicalPlan plan = plan("select * from " +
+                                " (select a, i from t1 order by a limit 5) t1 " +
+                                "left join" +
+                                " (select b, i from t2 where b > 10) t2 " +
+                                "on t1.i = t2.i where t1.a > 50 and t2.b > 100 " +
+                                "limit 10");
+        assertThat(plan, isPlan("FetchOrEval[a, i, b, i]\n" +
+                                "Limit[10;0]\n" +
+                                "HashJoin[\n" +
```

Comment:
Hash Join? with ``left`` :-)

Reply:
Seems to be due to our left join -> inner join rewrite - I copied the query from a AnalyzerTest and didn't think about it. 

Can change it if you want.

Reply:
@matriv pushed a fixup